nanofi/include/core/cstructs.h

```diff
@@ -149,4 +149,15 @@ typedef struct cstream {
   SOCKET socket_;
 } cstream;
 
+/****
+ * ##################################################################
+ *  STRING OPERATIONS
+ * ##################################################################
+ */
+
+typedef struct tokens {
```

Comment:
As this struct contains dynamically allocated memory, please provide a free_tokens function for it: it is less error-prone than manually freeing its members every time.